### ISSUE TITLE
Export atomic types as volatile

### ISSUE THREAD

**Issue by Tommoa** [eqrion/cbindgen#151]
Currently when trying to use an atomic type (e.g. `AtomicBool` or `AtomicUsize`), cbindgen exports them as their names. Instead, it should export as a `volatile` primitives.

Suggested mappings:
- `AtomicBool` -> `volatile char`
- `AtomicUsize` -> `volatile size_t`
- `AtomicIsize` -> `volatile ssize_t`
- `AtomicPtr<T>` -> `volatile T*`

**Comment by eqrion** [comment]
I don't believe that atomic types are equivalent to volatile qualified primitives. [1] Are you sure they're equivalent, or have any docs suggesting they are?

[1] http://en.cppreference.com/w/cpp/language/cv

**Comment by Tommoa** [comment]
@eqrion 

If we're exporting to C++, then of course C++ has is own atomic types. C11 does too (using the `_Atomic` qualifier), but anything earlier does not.

From the [cppreference section on the c `volatile` qualifier](http://en.cppreference.com/w/c/language/volatile):

> Note that volatile variables are not suitable for communication between threads; they do not offer atomicity, synchronization, or memory ordering. A read from a volatile variable that is modified by another thread without synchronization or concurrent modification from two unsynchronized threads is undefined behavior due to a data race. 

So yes, you're 100% correct that `volatile` is not an atomic type equivalent. It is, however, the only way in which C prior to C11 could declare atomics.

> Every access (both read and write) made through an lvalue expression of volatile-qualified type is considered an observable side effect for the purpose of optimization and is evaluated strictly according to the rules of the abstract machine (that is, all writes are completed at some time before the next sequence point). This means that within a single thread of execution, a volatile access cannot be optimized out or reordered relative to another visible side effect that is separated by a sequence point from the volatile access.

Ensuring that writes/reads do not get reordered (which `volatile` provides), is incredibly important for atomics. So while atomics operations are platform/library dependent, `volatile` is usually a good way to declare that a variable may be considered atomic.

Of course, if the option is available, using either the platform or library dependent types or `_Atomic` is preferable every time, but using `volatile` as a way to indicate an atomic is the next best way.

**Comment by eqrion** [comment]
> So yes, you're 100% correct that volatile is not an atomic type equivalent. It is, however, the only way in which C prior to C11 could declare atomics.

If `volatile T` is not type equivalent of `AtomicT` then cbindgen should not make that conversion. At best that would be misleading, at worst it'd be incorrect. Exposing atomics across FFI boundaries is non trivial, and I don't want to add a footgun here.
